testplan/testing/multitest/driver/connection/connection_extractor.py

```diff
@@ -0,0 +1,181 @@
+import socket
+import sys
+from typing import List
+import psutil
+
+from testplan.common.utils.logger import TESTPLAN_LOGGER
+from testplan.testing.multitest.driver.connection.base import (
+    BaseConnectionExtractor,
+    Direction,
+)
+from testplan.testing.multitest.driver.connection.connection_info import (
+    Protocol,
+    PortConnectionInfo,
+    FileConnectionInfo,
+)
+
+
+SOCKET_CONNECTION_MAP = {
+    socket.SocketKind.SOCK_STREAM: Protocol.TCP,
+    socket.SocketKind.SOCK_DGRAM: Protocol.UDP,
+}
+
+NETWORK_CONNECTION_MAP = {
+    Protocol.TCP: socket.SocketKind.SOCK_STREAM,
+    Protocol.UDP: socket.SocketKind.SOCK_DGRAM,
+}
+
+
+class ConnectionExtractor(BaseConnectionExtractor):
+    def __init__(self, service: str, protocol: Protocol, direction: Direction) -> None:
+        self.service = service
+        self.protocol = protocol
+        self.direction = direction
+
+    def extract_connection(self, driver) -> List[PortConnectionInfo]:
+
+        return [
+            PortConnectionInfo(
+                name="Port",
+                service=self.service,
+                protocol=self.protocol,
+                direction=self.direction,
+                identifier=driver.connection_identifier,
+                port=getattr(driver, "local_port", None),
+                host=getattr(driver, "local_host", None),
+            )
+        ]
+
+class PortConnectionExtractor(BaseConnectionExtractor):
```

Comment:
this extractor only works when driver has a subprocess, maybe change the class name to inform the user?